etc/scripts/set_copyright_year.sh

```diff
@@ -9,7 +9,36 @@
 #
 # The script will check the first two lines for a copyright
 # notice (in case the first line is a shebang).
+#
+# Run this script with --check to have it raise an error if it
+# change anything.
+
+
+# Set CHECK_MODE based on whether --check is passed
+    CHECK_MODE=false
+    if [[ "$1" == "--check" ]]; then
+        CHECK_MODE=true
+        shift # Remove --check from the arguments
+    fi
 
 while read -rd $'\0' year file; do
-    sed -i "1,2s/^\(# Copyright (c) \)[[:digit:]]\{4,\}/\1$year/" "$file"
+
+    # Extract the first year from the copyright notice
+    current_year=$(sed -n '1,2s/^\(# Copyright (c) \)\([[:digit:]]\{4,\}\).*/\2/p' "$file")
+
+    # Skip the file if no year is found
+    if [[ -z "$current_year" ]]; then
+        continue
+    fi
+
+    if $CHECK_MODE && [[ "$current_year" != "$year" ]]; then
+        echo "Error: Copyright year mismatch in file $file. Expected $year, found $current_year."
+        exit 1
```

Comment:
I think this is not quite enough. We have to compare the year found in the file with the year of the last commit the file has been modified in. (At least to my understanding of how the copyright notice works, please double check this.)

Reply:
The `current_year` is the year found in the file and the `year` is the year of the last commit for this file. I calculated `year` the same as before and also checked that it makes sense so it should be correct.